src/requirements/requirement-exam-mapping.ts

```diff
@@ -0,0 +1,122 @@
+import examData, {
+  DEFAULT_COLLEGES,
+  ExamFulfillment,
+  ExamFulfillments,
+} from './data/exams/ExamCredit';
+import { colleges, College } from './data';
+
+type ExamRequirementsColleges = Record<number, College[]>;
+type ExamRequirements = {
+  colleges?: ExamRequirementsColleges; // if the user IS NOT in one of these colleges, the course id cannot fulfill the requirement
+  majorsExcluded?: string[]; // if the user IS in one of these majors, the course id cannot fulfill the requirement
+};
+
+/**
+ * Flat list of exam fulfillments, i.e.
+ * [
+ *   { courseId: 100, minimumScore: 4, credits: 4 },
+ *   { courseId: 101, minimumScore: 5, credits: 8 },
+ *   {
+ *     courseId: 102,
+ *     courseEquivalents: { DEFAULT: [Array], EN: [Array] },
+ *     minimumScore: 5,
+ *     credits: 4
+ *   },
+ *   ...
+ * ]
+ */
+const examFulfillmentList: ExamFulfillment[] = Object.values(examData)
+  .map((examFulfillments: ExamFulfillments) => Object.values(examFulfillments))
+  .flat(2);
+
+/**
+ * Mapping from exam id to requirements info (course equivalents and exam conditions), i.e.
+ * {
+ *   "100": {},
+ *   "101": {},
+ *   "102": {
+ *     "colleges": {
+ *       "351265": [
+ *         "AG",
+ *         "AR",
+ *         "AS1",
+ *         "AS2",
+ *         "HE",
+ *         "IL",
+ *         "BU"
+ *       ],
+ *       "359187": [
+ *         "EN"
+ *       ]
+ *     }
+ *   },
+ *   ...
+ * }
+ */
+export const examRequirementsMapping: Record<number, ExamRequirements> = examFulfillmentList.reduce(
+  (mapping, fulfillment) => {
+    const { courseId, courseEquivalents, majorsExcluded } = fulfillment;
+
+    if (!courseEquivalents) {
+      // if no course equivalents, return empty mapping for exam
+      return { ...mapping, [courseId]: {} };
+    }
+
+    // for each id, assign a list of colleges for which the exam can fulfill requirements
+    const nonDefaultColleges = Object.keys(courseEquivalents);
```

Comment:
I see what you mean. The variable's value is correct but I think it can be renamed `definedColleges` or something. Whether or not the actual string `'DEFAULT'` is in there doesn't matter, since it's just being used as a mask over `colleges`.